src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -1235,15 +1283,21 @@ class SegmentCleaner : public SegmentProvider {
     } else {
       assert(segments[segment].is_empty());
       assert(segments.get_empty_segments(segment.device_id()) > 0);
-      segments.segment_used(segment, true);
-      stats.avail_bytes -= segments[segment.device_id()]->segment_size;
       stats.empty_segments--;
     }
-    crimson::get_logger(ceph_subsys_seastore).debug(
-      "mark_closed: device: {} empty_segments: {}",
-      segment.device_id(),
-      segments.get_empty_segments(segment.device_id()));
-    segments[segment].set_closed();
+    segments.segment_closed(segment);
+    stats.avail_bytes = segments.get_available_bytes(segment.device_id());
+    stats.opened_segments = segments.get_opened_segments(segment.device_id());
```

Comment:
The `avail_bytes` and `opened_segments` are at the `SegmentCleaner` level for all the managed devices, right?

But seems here (and below) they are assigned to be the stats by a specific device.

Reply:
ah, yes